napari/_qt/pyhon_syntax_highlight.py

```diff
@@ -0,0 +1,68 @@
+from pygments import highlight
+from pygments.formatter import Formatter
+from pygments.lexers import get_lexer_by_name
+from pygments.styles import get_style_by_name
+from qtpy import QtGui
+
+# https://github.com/Vector35/snippets/blob/master/QCodeEditor.py
+
+
+def _format(style):
+    """Return a QTextCharFormat with the given attributes."""
+    _format = QtGui.QTextCharFormat()
+    if "#" in style:
+        index = style.index("#")
+        color = style[index : index + 7]
+        _color = QtGui.QColor(color)
+
+        _format.setForeground(_color)
+
+    if "bold" in style and "nobold" not in style:
+        _format.setFontWeight(QtGui.QFont.Bold)
+    if "italic" in style and "noitalic" not in style:
+        _format.setFontItalic(True)
+    if "underline" in style and "nounderline" not in style:
+        _format.setFontUnderline(True)
+
+    return _format
+
+
+class QFormatter(Formatter):
+    def __init__(self, theme):
+        super().__init__()
+        self._theme = theme
+        self._style = {
+            str(name): _format(style)
+            for name, style in get_style_by_name(self._theme).styles.items()
+        }
+
+    def format(self, tokensource, outfile):
+        self.data = []
+
+        for token, value in tokensource:
+            self.data.extend(
+                [
+                    self._style[str(token)],
+                ]
+                * len(value)
+            )
```

Comment:
I find it confusing that `outfile` is not used at all, and instead an attribute is created (?) and then extended. Documentation would help here also.